### PR TITLE
A configurator for anime-dl

### PR DESCRIPTION
**Pull request by Arjix-2nd**
Huge thx to Lagrad for helping me sooooo much, originally i made this but lagrad rewrote it and made things so much easier for me.
This sub-command covers the basic configurations that anime-dl has. i will list them down below:
Quality, Provider, Download directory, Skip Download, Player, File format, external_downloader, chunk size, aria2 for torrents.
I have also added options for selescrape, they are commented off now cause selescrape isnt approved yet.

### CODE REVIEW

File: anime_downloader/commands/config.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import json
+import os
+import sys
+import click
+import time
+import logging
+from tabulate import tabulate
+from anime_downloader.sites import ALL_ANIME_SITES
+sitenames = [v[1] for v in ALL_ANIME_SITES]
+def clear():
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:10]: Space

File: anime_downloader/commands/config.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import json
+import os
+import sys
+import click
+import time
+import logging
+from tabulate import tabulate
+from anime_downloader.sites import ALL_ANIME_SITES
+sitenames = [v[1] for v in ALL_ANIME_SITES]
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:9]: space

File: anime_downloader/commands/config.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+def clear():    
+    os.system('cls||clear')
+def get_config(): #can't import config directly because of circular import
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:12]: space

File: anime_downloader/commands/config.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        if key in d:
+            d = d[key]
+        elif create_missing:
+            d = d.setdefault(key, {})
+        else:
+            return dic
+    if keys[-1] in d or create_missing:
+        d[keys[-1]] = value
+    return dictionary
+
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:48]: You have to use `click.command`. Look at `dl` for an example

File: anime_downloader/commands/config.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+layout = {
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:49]: I don't like copying over the config. You have to import the DEFAULT_SETTINGS from config.py

File: anime_downloader/commands/config.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import os
+import sys
+import click
+import time
+import logging
+from tabulate import tabulate
+from anime_downloader.sites import ALL_ANIME_SITES
+sitenames = [v[1] for v in ALL_ANIME_SITES]
+def clear():    
+    os.system('cls||clear')
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:11]: Using `click.clear`

**Comment by AbdullahM0hamed** [comment]
It should be done now

File: anime_downloader/commands/config.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        if type(newVal) != type(data[keys[val]]):
+            choice = click.confirm(f"{newVal} appears to be of an incorrect type. Continue")
+
+            if not choice:
+                exit()
+
+        data[keys[val]] = newVal
+
+@click.command()
+def command():
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:40]: Needs doc strings

Comment (AbdullahM0hamed) [reply anime_downloader/commands/config.py]: Done

File: anime_downloader/commands/config.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+def create_table(_list, previous):
+    newList = [(x, y) for x, y in zip(range(1,len(_list) + 1), _list)]
+    headers = ['SlNo', f'{previous} settings'.strip()]
+    table = tabulate(newList, headers, tablefmt = "psql")
+    table = "\n".join(table.split("\n")[::-1])
+    return table
+
+def traverse_json(data, previous=''):
+    click.clear()
+    keys = [*data.keys()]
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:16]: `list(data.keys())`?

File: anime_downloader/commands/config.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import click
+from tabulate import tabulate
+from anime_downloader.config import Config
+
+data = Config._CONFIG
+
+def create_table(_list, previous):
+    newList = [(x, y) for x, y in zip(range(1,len(_list) + 1), _list)]
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:8]: `zip(enumerate(_list))`?

File: anime_downloader/commands/config.py [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        if type(newVal) != type(data[keys[val]]):
+            choice = click.confirm(f"{newVal} appears to be of an incorrect type. Continue")
+
+            if not choice:
+                exit()
+            else:
+                try:
+                    newVal = type(data[keys[val]])(newVal)
+                except TypeError:
+                    click.prompt(f"'{newVal}' could not be converted to the correct type")
```
Comment (vn-ki) [inline anime_downloader/commands/config.py:41]: Change from `prompt` to `echo` or i think `error`?

**Pull request merged**